packages/babel-core/src/transformation/file/index.js

```diff
@@ -246,7 +246,12 @@ export default class File extends Store {
       name,
     ));
 
-    const { nodes, globals } = getHelper(name, uid, ownBindingNames);
+    const { nodes, globals } = getHelper(
+      name,
+      name => this.addHelper(name),
```

Comment:
I try to avoid using sideeffectful functions like `addHelper` when writing utils that focus building and retrieving data like `getHelper` since `get*` as a name doesn't really make you expect the function to have sideeffects.

An alternate approach that comes to mind here would be that `getHelpers` could accept a callback like this that just returns the name of the helper, if it exists, or null. Then `getHelper` could be adjusted to return _multiple_ helpers, and include the ones that don't exist yet in that list.

Thoughts?

Reply:
Actually `getHelper` doesn't have side effects: the `loadDependency` parameter is just "a function which, given an helper name, returns a node which references the helper": the side effects are in that that callback, but I think that it is easily recognizeable because its defined inside the `addHelper` function (which _has_ side effects).

For example, in `babel-core/src/tools/babel-external-helpers.js` it is used in a side effects free way.

Reply:
I was being grumpy about it because it technically means that the `ownBindingNames` passed into the `getHelper` will be outdated since it is calculated before the `addHelper` call for the dependencies. If the dependencies happened to name something that conflicted with the helper that depends on it, they'd potentially conflict.

That said, it's probably an edge case that isn't worth worrying about.

Reply:
I fixed that possible bug by lazy-computing `ownBindingNames` (so that they are retrieved after the dependencies are added).